### PR TITLE
Update governance model

### PR DESCRIPTION
**Pull request by christian-bromann**
This governance change will better describe what roles there are and how to get from one role to another.

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -1,66 +1,119 @@
...
 
-* Any two TC members can decide to make someone a project committer, providing that person has shown an appropriate understanding of the WebdriverIO codebase through prior contributions of code.
-* In general any committer can review and merge a PR. In general committers should only merge code they are qualified to review, which might entail pinging another committer who has greater ownership over a specific code area.
+Committers:
+
+* Are expected to work on public branches of the source repository and submit pull requests from that branch to the master branch.
+* Proposals for large changes to the project's code (architectural changes, etc...) should be brought forward as a GitHub issue (with the label Proposal), and all committers should be pinged so they can weigh in on the discussion if desired. Substantial changes, whether in proposal stage or in pull request stage should be signed off on by 1 TC member and at least 2 other committers. To assist in the discussion, a small proof of concept can be undertaken on a subset of the WebdriverIO ecosystem and raised as a strawman PR, to give flesh to the proposal and make discussion more helpfully concrete.
```
Comment (WillBrock) [inline GOVERNANCE.md]: Change the TC to TSC.

File: GOVERNANCE.md [hunk 2]
```diff
@@ -1,66 +1,119 @@
...
+If an agenda item cannot reach a consensus, a TSC member can call for either a closing vote or a vote to table the issue to the next meeting. The call for a vote must be approved by a majority of the TSC or else the discussion will continue. Simple majority wins.
+
+## Sponsorship
+
+The WebdriverIO project welcomes sponsorship of various organizations. Specific details are not yet available but in general we will accept monetary donations, contribution of code, contribution of dedicated committers, or donations of other goods or service (internet hosting, etc...). To begin the conversation, reach out to one of the TSC members.
+
+## Raising Issues Related to Governance
 
 This governance model necessarily leaves many situations unspecified. If questions arise as to how a given situation should proceed according to the overall goals of the project, the best thing to do is to open a GitHub issue and ping the TC members.
+
```
Comment (WillBrock) [inline GOVERNANCE.md]: > the best thing to do is to open a GitHub issue and ping the TC members

TSC

File: GOVERNANCE.md [hunk 3]
```diff
@@ -1,66 +1,119 @@
...
+#### Process for Adding Committers
+
+1. Add the GitHub user to the "Project Committers" team
+2. Invite to Gitter team chatroom (`webdriverio/ProjectCommitters`)
+3. Tweet congratulations to the new committer from the WebdriverIO Twitter account
+
+### Technical Steering Committee (TSC)
+
+The WebdriverIO project is jointly governed by a Technical Committee (TSC) which is responsible for high-level guidance of the project.
```
Comment (WillBrock) [inline GOVERNANCE.md]: Technical Steering Committee instead of Technical Committee

File: GOVERNANCE.md [hunk 4]
```diff
@@ -1,66 +1,119 @@
...
+* Have given good feedback on others' submissions and displayed an overall understanding of the code quality standards for the project.
+* Commit to being a part of the community for the long-term.
+* Have submitted a minimum of 20 qualifying pull requests.
+
+A Committer is invited to become a TSC member by existing TSC members. A nomination will result in discussion and then a decision by the TSC.
+
+#### Process for Adding TSC Members
+
+1. Add the GitHub user to the "WebdriverIO TSC" team
+1. Set the GitHub user to be have the "Owner" role for the WebdriverIO organization
```
Comment (WillBrock) [inline GOVERNANCE.md]: Remove "be"

**Comment by christian-bromann** [comment]
@WillBrock @abjerstedt can I get a final review?

File: GOVERNANCE.md [hunk 5]
```diff
@@ -1,66 +1,119 @@
...
 
-* Membership in the TC is granted by unanimous vote of current TC members.
-* TC members can step down at any time.
-* No more than 1/3 of TC members may belong to the same company.
-* Changes to project governance require unanimous vote of the TC.
+### Contributors
+
+Contributors are community members who contribute in concrete ways to the project, most often in the form of code and/or documentation. Anyone can become a Contributor, and contributions can take many forms. There is no expectation of commitment to the project, no specific skill requirements, and no selection process.
+
+Contributors have read-only access to source code and so submit changes via pull requests. Contributor pull requests have their contribution reviewed and merged by a TSC member. TSC members and Committers work with Contributors to review their code and prepare it for merging.
```
Comment (erwinheitzman) [inline GOVERNANCE.md]: Contributors have read-only access to source code and to submit changes via pull requests.

File: GOVERNANCE.md [hunk 6]
```diff
@@ -1,66 +1,119 @@
...
+* Are expected to delete their public branches when they are no longer necessary.
+* Must submit pull requests for all changes.
+* Have their work reviewed by TSC members before acceptance into the repository.
+* May label and close issues
+* May merge some pull requests
 
-### Contributors
+To become a Committer:
+
+* One must have shown a willingness and ability to participate in the project as a team player. Typically, a potential Committer will need to show that they have an understanding of and alignment with the project, its objectives, and its strategy.
```
Comment (erwinheitzman) [inline GOVERNANCE.md]: it's objectives, and it's strategy.

File: GOVERNANCE.md [hunk 7]
```diff
@@ -1,66 +1,119 @@
...
+* Committers are expected to be respectful of every community member and to work collaboratively in the spirit of inclusion.
+* Have submitted a minimum of 10 qualifying pull requests. What's a qualifying pull request? One that carries significant technical weight and requires little effort to accept because it's well documented and tested.
+
+New Committers can be nominated by any existing Committer. Once they have been nominated, there will be a vote by the TSC members.
+
+It is important to recognize that committership is a privilege, not a right. That privilege must be earned and once earned it can be removed by the TSC members by a standard TSC motion. However, under normal circumstances committership exists for as long as the Committer wishes to continue engaging with the project.
+
+A Committer who shows an above-average level of contribution to the project, particularly with respect to its strategic direction and long-term health, may be nominated to become a TSC member, described below.
```
Comment (erwinheitzman) [inline GOVERNANCE.md]: with respect to it's

File: GOVERNANCE.md [hunk 8]
```diff
@@ -1,66 +1,119 @@
...
+The WebdriverIO Project wants as much as possible to operate using procedures that are fair, open, inviting, and ultimately good for the community. For that reason we find it valuable to codify some of the ways that the Project goes about its day-to-day business. We want to make sure that no matter who you are, you have the opportunity to contribute to WebdriverIO. We want to make sure that no corporation can exert undue influence on the community or hold the Project hostage. And likewise we want to make sure that corporations which benefit from WebdriverIO are also incentivized to give back. This document describes how various types of contributors work within the WebdriverIO project.
```
Comment (erwinheitzman) [inline GOVERNANCE.md]: it's day-to-day business

File: GOVERNANCE.md [hunk 9]
```diff
@@ -1,66 +1,119 @@
...
+### Users
 
-* [@abjerstedt](https://github.com/abjerstedt)
-* [@bromann](https://github.com/christian-bromann)
-* [@willbrock](https://github.com/willbrock)
+Users are community members who have a need for the project. Anyone can be a User; there are no special requirements. Common User contributions include evangelizing the project (e.g., display a link on a website and raise awareness through word-of-mouth), informing developers of strengths and weaknesses from a new user perspective, or providing moral support (a "thank you" goes a long way).
 
-They all represent different companies and different types of contribution to the WebdriverIO Project. The TC is responsible for high-level decisions (like [release a new version](/CONTRIBUTING.md#release-new-version), establishing a feature roadmap, accepting sponsorship, organizing conferences, etc...), adjudicating conflicts between members of the community, and making changes to the project's governance processes.
+Users who continue to engage with the project and its community will often become more and more involved. Such Users may find themselves becoming Contributors, as described in the next section.
```
Comment (erwinheitzman) [inline GOVERNANCE.md]: it's community

File: GOVERNANCE.md (hunk 6 above)
Comment (WillBrock) [reply GOVERNANCE.md]: `its objectives, and its strategy` is correct in this case.

File: GOVERNANCE.md (hunk 7 above)
Comment (WillBrock) [reply GOVERNANCE.md]: `its` is correct here as well.

File: GOVERNANCE.md (hunk 8 above)
Comment (WillBrock) [reply GOVERNANCE.md]: `its` is correct here as well.

File: GOVERNANCE.md (hunk 9 above)
Comment (WillBrock) [reply GOVERNANCE.md]: `its` is correct here as well.

File: GOVERNANCE.md (hunk 5 above)
Comment (WillBrock) [reply GOVERNANCE.md]: Updating `so` to `to` and then I think the pr is good to go.

File: GOVERNANCE.md [hunk 10]
```diff
@@ -1,66 +1,119 @@
...
-* [@lgandecki](https://github.com/lgandecki)
-* [@tborys](https://github.com/tborys)
-* [@TheSavior](https://github.com/TheSavior)
-* [@ThomasFett](https://github.com/ThomasFett)
-* [@TuHuynhVan](https://github.com/TuHuynhVan)
-* [@WillBrock](https://github.com/WillBrock)
-* [@wswebcreation](https://github.com/wswebcreation)
-* [@wvankuipers](https://github.com/wvankuipers)
-* [@zinserjan](https://github.com/zinserjan)
+* In general any committer can review and merge a PR. In general committers should only merge code they are qualified to review, which might entail pinging another committer who has greater ownership over a specific code area.
```
Comment (abjerstedt) [inline GOVERNANCE.md]: This contradicts the lines above about TSC members merging, and the requirement for 3 approvals, and the line below about work reviewed by TSC members.

File: GOVERNANCE.md [hunk 11]
```diff
@@ -1,66 +1,119 @@
...
-### Contributors
+To become a Committer:
+
+* One must have shown a willingness and ability to participate in the project as a team player. Typically, a potential Committer will need to show that they have an understanding of and alignment with the project, its objectives, and its strategy.
+* Committers are expected to be respectful of every community member and to work collaboratively in the spirit of inclusion.
+* Have submitted a minimum of 10 qualifying pull requests. What's a qualifying pull request? One that carries significant technical weight and requires little effort to accept because it's well documented and tested.
+
+New Committers can be nominated by any existing Committer. Once they have been nominated, there will be a vote by the TSC members.
+
+It is important to recognize that committership is a privilege, not a right. That privilege must be earned and once earned it can be removed by the TSC members by a standard TSC motion. However, under normal circumstances committership exists for as long as the Committer wishes to continue engaging with the project.
```
Comment (abjerstedt) [inline GOVERNANCE.md]: missing comma after "once earned"

File: GOVERNANCE.md [hunk 12]
```diff
@@ -1,66 +1,119 @@
...
 
-* Any two TC members can decide to make someone a project committer, providing that person has shown an appropriate understanding of the WebdriverIO codebase through prior contributions of code.
-* In general any committer can review and merge a PR. In general committers should only merge code they are qualified to review, which might entail pinging another committer who has greater ownership over a specific code area.
+Committers:
+
+* Are expected to work on public branches of the source repository and submit pull requests from that branch to the master branch.
+* Proposals for large changes to the project's code (architectural changes, etc...) should be brought forward as a GitHub issue (with the label Proposal), and all committers should be pinged so they can weigh in on the discussion if desired. Substantial changes, whether in proposal stage or in pull request stage should be signed off on by 1 TSC member and at least 2 other committers. To assist in the discussion, a small proof of concept can be undertaken on a subset of the WebdriverIO ecosystem and raised as a strawman PR, to give flesh to the proposal and make discussion more helpfully concrete.
```
[the remark this answers is not in the document]
Comment (christian-bromann) [reply GOVERNANCE.md]: not sure if we can enforce this but I can add a section to this

File: GOVERNANCE.md (hunk 10 above)
Comment (christian-bromann) [reply GOVERNANCE.md]: I think this section is about bug fixes, doc changes or other smaller contributions. A requirement for 3 approvals only applies for substantial changes, e.g. features like spec level retries

File: GOVERNANCE.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (christian-bromann) [reply GOVERNANCE.md]: Changed the first committers section as follows:

> * Are expected to work on public branches of the source repository and submit pull requests from that branch to the master branch. Pull requests has to be made as a draft if the work is not ready for a merge nor an initial review.

File: GOVERNANCE.md [hunk 13]
```diff
@@ -24,7 +24,7 @@ Committers are community members who have shown that they are committed to the c
 
 Committers:
 
-* Are expected to work on public branches of the source repository and submit pull requests from that branch to the master branch.
+* Are expected to work on public branches of the source repository and submit pull requests from that branch to the master branch. Pull requests has to be made as a draft if the work is not ready for a merge nor an initial review.
```
Comment (abjerstedt) [inline GOVERNANCE.md]: "Pull requests *have* to be" or "*A* pull request has to be be"

**Comment by christian-bromann** [comment]
@abjerstedt any objections to merge this?

**Pull request merged**